shapely/tests/__init__.py

```diff
@@ -0,0 +1,24 @@
+import os
+import sys
+import unittest  # NOQA
+
+if os.name == "nt" and sys.version_info[1] >= 8:
+    geos_path = os.environ.get("GEOS_INSTALL")
+    if geos_path:
+        os.add_dll_directory(geos_path + r"\bin")
+
+
+import numpy
+import pytest
+
+from shapely.geos import geos_version_string
+
+# Show some diagnostic information; handy for CI
```

Comment:
This should be consolidated into [pytest_report_header](https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_report_header) in `conftest.py`

Reply:
I'm thinking to leave this to another PR as it seems out of scope ?

Reply:
Follow-on PRs are just fine for these changes, in order to keep this one focused purely on moving & formatting code.  They just jumped out as needing to be fixed at some point.  😄

Reply:
Can you keep this `__init__.py` in the legacy subdirectory? And the same for the conftest.py file? 

Both are not needed for the existing /shapely/tests/ to run, and as Brendan mentioned above, this content actually belongs elsewhere. So until that is done (which indeed doesn't need to be in this PR), let's leave it as is in the /legacy subdirectory